src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -811,6 +832,192 @@ public static Collection<PSObject> InvokeOnRunspace(PSCommand command, Runspace
 
         #endregion
 
+        #region Feedback Rendering
+
+        /// <summary>
+        /// Render the feedbacks to the specified host.
+        /// </summary>
+        /// <param name="feedbacks">The feedback results.</param>
+        /// <param name="ui">The host to render to.</param>
+        public static void RenderFeedback(List<FeedbackResult> feedbacks, PSHostUserInterface ui)
+        {
+            // Caption style is dimmed bright white with italic effect, used for fixed captions, such as '[' and ']'.
+            string captionStyle = "\x1b[97;2;3m";
+            string italics = "\x1b[3m";
+            string nameStyle = PSStyle.Instance.Formatting.FeedbackName;
+            string textStyle = PSStyle.Instance.Formatting.FeedbackText;
+            string actionStyle = PSStyle.Instance.Formatting.FeedbackAction;
+            string ansiReset = PSStyle.Instance.Reset;
+
+            if (!ui.SupportsVirtualTerminal)
+            {
+                captionStyle = string.Empty;
+                italics = string.Empty;
+                nameStyle = string.Empty;
+                textStyle = string.Empty;
+                actionStyle = string.Empty;
+                ansiReset = string.Empty;
+            }
+
+            var output = new StringBuilder();
+            var chkset = new HashSet<FeedbackItem>();
+
+            foreach (FeedbackResult entry in feedbacks)
+            {
+                output.AppendLine();
+                output.Append(captionStyle).Append('[').Append(ansiReset)
+                    .Append(nameStyle).Append(italics).Append(entry.Name).Append(ansiReset)
+                    .Append(captionStyle).Append(']').Append(ansiReset);
```

Comment:
We could use interpolated string handler overload.

Reply:
Updated with interpolated string handler.